### ISSUE TITLE
Completing the 2nd, 3rd, 4th [...] occurence of a repeating task via notification doesn't set the task status to completed

### ISSUE THREAD

**Issue by hundertprobert** [FossifyOrg/Calendar#156]
### Checklist

- [X] I can reproduce the bug with the latest version given [here](https://github.com/FossifyOrg/Calendar/releases/latest).
- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Calendar/issues) or [closed](https://github.com/FossifyOrg/Calendar/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Calendar/discussions) or [closed](https://github.com/FossifyOrg/Calendar/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Calendar/blob/master/CONTRIBUTING.md).

### Affected app version

1.0.2

### Affected Android/Custom ROM version

Android 10 / MIUI Global 12.0.10

### Affected device model

Xiaomi Mi 9 Lite

### How did you install the app?

F-Droid / IzzyOnDroid

### Steps to reproduce the bug

1. Create a daily repeating task with notification at the starting time
2. Set it as completed on its first occurence 
3. Receive notification for the following occurence (= on the next day)
4. within the notification, click on "mark as completed"

### Expected behavior

The task should be marked as completed for this day when opening the calendar app and in the widget view.

### Actual behavior

The task remains as uncompleted in the calendar view as well as the widget view, despite being marked as completed via notification. 

### Additional information

This bug was also reported by users in a discussion of a separate bug, see here: https://github.com/FossifyOrg/Calendar/issues/26#issuecomment-1912057605 and here: https://github.com/FossifyOrg/Calendar/issues/26#issuecomment-1912060364

**Comment by Aga-C** [comment]
Can you reproduce it every time with steps you've written? Because I've tested it on three phones, and every time it works. However, it was always the first occurrence, when in one of the comments you've linked, the problem was for third and further.

Also, is the problem only visual on a widget and calendar view, or is it also incomplete when you go to the task editing view?

**Comment by hundertprobert** [comment]
You're right, I just tested it as well with a new task and changed the description above. The first occurence works just fine, the following does not. I've had this happen with a long-running task (a daily reminder) where I then discovered there were 4 or 5 uncompleted occurences in the calendar view that I definitely marked as completed. 

> Also, is the problem only visual on a widget and calendar view, or is it also incomplete when you go to the task editing view?

The task is also incomplete in the task editing view.

**Comment by MrChocolatine** [comment]
I also experienced this problem, even when I was using the SimpleMobileTools version. (so for a very long time)

**Comment by eggsnpie** [comment]
I've also had this issue even with the first occurrence of an event/task. When I click to mark them completed in the notification, it doesn't. It just acts as if I dismissed them.

Android 14
Pixel 8

**Comment by eggsnpie** [comment]
This still does happen in the newest versions as well.

**Comment by naveensingh** [comment]
Thanks for the ping.

I'm well aware of the issue and the stakes involved. I do read all the messages here and this issue _is_ on my TODO list.

As for notifying users, there are many open issues that may potentially cause harm to someone somewhere so we cannot add all of them to the README and it probably won't help much anyway since there are many users don't know what GitHub is. I have started adding an "About => Known issues" option in every app for the same reason.

In the meantime, please avoid using Fossify Calendar for things like medications.

**Comment by MrChocolatine** [comment]
> to be blatant: someone's gonna OD on their meds because of this, so it's _quite_ dangerous and irresponsible to keep things as they are

Are you for real? Instead of blaming the app and the author, just take your own responsibility and use another app ffs.
